src/Sentry.Maui/Internal/SentryMauiScreenshotProcessor.cs

```diff
@@ -18,6 +18,9 @@ public SentryMauiScreenshotProcessor(SentryMauiOptions options)
 
     public SentryEvent? Process(SentryEvent @event, SentryHint hint)
     {
+        // Call back before taking the screenshot if the callback is mentioned and if a Screenshot is attached
+        _options.BeforeCaptureScreenshot();
```

Comment:
We probably want to validate the return value here to allow opting-out of the screenshot taking. Otherwise the callback doesn't do much. 
That way users gain some control - i.e. they know that they do not want a screenshot taken due to PII at that exact moment.

Reply:
I think you're right! 
In my head I thought they would validate the return while setting the callback, but it makes very little sense, indeed, in retrospect.
Thanks, I'll do that.